pkg/model/record.go

```diff
@@ -97,16 +107,72 @@ func NewRecord(
 			record.DNSLatency = time.Duration(metrics.AdditionalMetrics.DnsRecord.Latency)
 		}
 	}
-	record.NetworkMonitorEventsMD = make([]config.GenericMap, 0)
+	if s != nil && metrics.AdditionalMetrics != nil {
+		seen := make(map[string]bool)
+		record.NetworkMonitorEventsMD = make([]map[string]string, 0)
+		for _, metadata := range metrics.AdditionalMetrics.NetworkEvents {
+			if !AllZerosMetaData(metadata) {
+				var cm map[string]string
+				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
+					acl, ok := md.(*ovnmodel.ACLEvent)
+					mdStr := md.String()
+					if !seen[mdStr] {
+						if ok {
+							cm = map[string]string{
+								"Action":    acl.Action,
+								"Type":      acl.Actor,
+								"Feature":   "acl",
+								"Name":      acl.Name,
+								"Namespace": acl.Namespace,
+								"Direction": acl.Direction,
+							}
+						} else {
+							cm = map[string]string{
+								"Message": mdStr,
+							}
+						}
+						record.NetworkMonitorEventsMD = append(record.NetworkMonitorEventsMD, cm)
+						seen[mdStr] = true
+					}
+				}
+			}
+		}
+	}
 	return &record
 }
 
-type IntfDir struct {
+type IntfDirUdn struct {
 	Interface string
 	Direction int
+	Udn       string
 }
 
-func NewIntfDir(intf string, dir int) IntfDir { return IntfDir{Interface: intf, Direction: dir} }
+func NewIntfDirUdn(intf string, dir int, s *ovnobserv.SampleDecoder) IntfDirUdn {
+	var udn string
+	if s == nil {
+		return IntfDirUdn{Interface: intf, Direction: dir, Udn: ""}
+	}
+
+	// Load UDN cache if empty
+	if len(udnsCache) == 0 {
+		m, err := s.GetInterfaceUDNs()
+		if err != nil {
+			return IntfDirUdn{Interface: intf, Direction: dir, Udn: ""}
+		}
+		udnsCache = m
```

Comment:
this cache is never updated? I can add that also as a follow-up if you want

Reply:
well the caches is reset every eviction then read from ovnk after that its just ready only no update is needed during the eviction processing

Reply:
oh yes I didn't noticed in NewRecord